### PR TITLE
Add centerloss and refactor symbf

### PR DESCRIPTION
**Pull request by PatWie**
symbolic functions.py should be slim as for most projects there is no
need for the entire collection of embedding losses.

### CODE REVIEW

**Comment by ppwwyyxx** [comment]
Don't know exactly why, but running `gm convert results.jpg new.jpg` makes the image half the size without visible quality change (https://github.com/ppwwyyxx/tensorpack/pull/400/commits/12648f22c69393447cb2f1aab4746ce8a76b65fe)

**Comment by PatWie** [comment]
I used opencv to combine the old and new one. Too lazy to run all algorithms again.

**Comment by Skylion007** [comment]
Ah man, I was using those symbolic functions. They are actually incredibly useful for any type of Siamese networks, I would recommend commonly used metrics/model types should be better stored in Tensorpack. For instance, does GAN.py with it's useful and specific trainers really belong in examples?

**Comment by ppwwyyxx** [comment]
Apologies for the break. I should've marked them as deprecated first.
They're removed for the same reasons why GAN is not included or [freeze graph](https://github.com/ppwwyyxx/tensorpack/issues/386#issuecomment-324694820) is not included. Being useful is never the whole reason to add something.

**Comment by PatWie** [comment]
I started a `loss.py` file. But I do not want to maintain it and seems to be not in line with the decision to use `tf.layers`. But it might be worth at thinking about such a `loss.py` file as TensorFlow only supports streaming-metrics (not loss functions).

**Comment by ppwwyyxx** [comment]
TF has a few in `tf.losses.xxx` and a few in `tf.nn.xxx_loss`. But yes they are very limited.
However I do need to remove `huber_loss` and use `tf.losses.huber_loss`.

Having some symbolic function collections is never a bad idea but tensorpack isn't the right place for it. I'd like to see a project called "a bunch of useful TF symbolic functions" and then not only tensorpack but also anyone who uses tensorflow could benefit from it.

**Comment by Skylion007** [comment]
Alternatively, someone could create that project and have it be a dependency of Tensorpack. While Tensorpack may not be the proper place for many of these loss functions, it has by far some of the cleanest and best implementations of them. I'd be happy to try to start that project if there is interest. @PatWie care to share a gist of the loss.py file?

Regardless, I wouldn't mind seeing loss.py as part of say the example section.
